include/giskard_core/yaml_parser.hpp

```diff
@@ -1596,6 +1596,49 @@ namespace YAML {
     }
   };
 
+  //
+  // parsing of alias reference specs
+  //
+
+  inline bool is_alias_reference(const Node& node)
+  {
+    if(!node.IsScalar() || is_const_double(node))
```

Comment:
I'm surprised that this works. Am I misunderstanding the condition here? I read it as "If the node is either not a scalar or is a constant double, then it is not an alias." But shouldn't it just be "If the node is a scalar, then it's not an alias"? 
The unit tests seems to pass and this really baffles me.